src/dataflow/src/sink/kafka.rs

```diff
@@ -626,6 +679,34 @@ impl KafkaSinkState {
     }
 
     async fn flush(&self) -> KafkaResult<()> {
+        loop {
+            self.flush_inner().await?;
+            if {
+                let mut guard = self.retry_manager.lock().unwrap();
+                guard.sends_flushed()
+            } {
+                break;
+            }
```

Comment:
```suggestion
       self.flush_inner().await?;
        while !{
                let mut guard = self.retry_manager.lock().unwrap();
                guard.sends_flushed()
            } {
         self.flush_inner().await?;
```
Something like this might be cleaner? idk